src/brsTypes/components/RoSGNode.ts

```diff
@@ -251,18 +324,24 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
         return this.getMethod(index.value) || BrsInvalid.Instance;
     }
 
-    set(index: BrsType, value: BrsType, alwaysNotify: boolean = false) {
+    set(index: BrsType, value: BrsType, alwaysNotify: boolean = false, kind?: FieldKind) {
         if (index.kind !== ValueKind.String) {
             throw new Error("RoSGNode indexes must be strings");
         }
+
+        let fieldType = kind || FieldKind.fromBrsType(value);
+        // RBI does not set a field if the new value isn't valid
```

Comment:
As far as I could tell from playing around with it in RBI, this was true for all of the functions that could set a field's value: `addField`, `addFields`, `update`, `setField`, `setFields`. If you're trying to add a field (i.e. `update` or similar), and the value is `invalid`, the field won't get added at all. If you're trying to set a new value on an existing field (`myNode.foo = invalid`), then it won't set the new value to `invalid`.

Buuut if anyone has counter-evidence pls let me know

Reply:
ok so it turns out you _can_ set a `node`, `assocarray`, or `array` to invalid in RBI. Let me update this